pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -480,6 +485,27 @@ func (e *eksOperatorController) generateAndSetServiceAccount(cluster *mgmtv3.Clu
 	return e.clusterClient.Update(cluster)
 }
 
+func (e *eksOperatorController) canUsePublicAPIEndpoint(cluster *mgmtv3.Cluster) (string, *bool) {
```

Comment:
If you are generating the SA token here then the name should be changed to more accurately reflect what is happening.

Reply:
Yes, that is a bad name.